packages/story-editor/src/components/carousel/pagepreview/index.js

```diff
@@ -101,10 +101,19 @@ function PagePreview({ page, label, ...props }) {
   const [imageBlob, setImageBlob] = useState();
   const [pageNode, setPageNode] = useState();
   const setPageRef = useCallback((node) => node && setPageNode(node), []);
-  const hasImage = !isActive && imageBlob;
   const pageAtGenerationTime = useRef();
   const enableThumbnailCaching = useFeature('enableThumbnailCaching');
 
+  const activeStateTracker = useRef(isActive);
+  const hasImage = (!isActive || !activeStateTracker.current) && imageBlob;
+
+  useEffect(() => {
+    // We're keeping track of the previous active state
+    // so that if the page was not active before but is now, an image would be displayed
+    // since this means that the page hasn't changed compared to its image.
+    activeStateTracker.current = isActive;
+  }, [isActive]);
+
   // Whenever the page is re-generated and this is not the active page
   // remove the old (and now stale) image blob
   useEffect(() => {
```

Comment:
Actually, thinking a bit more about it, wouldn't just removing the `&& !isActive` check from the line below do the trick? Then the blob won't be cleared just because the page is made active - only once the page actually starts changing.

Reply:
You mean removing `&& !isActive` from below and displaying a thumbnail whenever `imageBlob` is truthy?

Reply:
Yes, I think that would work - because the image blob is only removed when the page is changed and only regenerated once that page is inactive again.